tests/Helper/HeaderHelperTest.php

```diff
@@ -134,37 +150,75 @@ public function testSortedAcceptTypesFromRequest(RequestInterface $request, arra
-            'oneQuotedSpace1' => ['a ="test"', ['a' => 'test']],
-            'oneQuotedSpace2' => ['a= "test"', ['a' => 'test']],
-            'oneQuotedSpace3' => ['a = "test"', ['a' => 'test']],
-            'semicolonAtEnd' => ['a = b;', ['a' => 'b']],
-            'semicolonAndSpaceAtEnd' => ['a = b; ', ['a' => 'b']],
-            'mixedQuotes' => ['a="test\'";test = "\'test\'"', ['a' => 'test\'', 'test' => '\'test\'']],
-            'specChars' => ['a=!#$%&\'*+.^`|~-; b=test', ['a' => '!#$%&\'*+.^`|~-', 'b' => 'test']],
-            'numbers' => ['a=8888;b="999"', ['a' => '8888', 'b' => '999']],
-            'invalidQuotes2' => ['a="test', null, \InvalidArgumentException::class],
-            'invalidQuotes3' => ['a=test"', null, \InvalidArgumentException::class],
-            'invalidEmptyQuotes' => ['a=""', null, \InvalidArgumentException::class],
-            'invalidEmptyValue' => ['a=b; c=', null, \InvalidArgumentException::class],
+            'simple' => ['a=test;test=test55',null, null, ['a' => 'test', 'test' => 'test55']],
+            'quoted' => ['a="test";b="test2;";d="."',null, null, ['a' => 'test', 'b' => 'test2;', 'd' => '.']],
+            'mixed' => ['a=b;c="apple"',null, null, ['a' => 'b', 'c' => 'apple']],
+            'one' => ['a=test',null, null, ['a' => 'test']],
+            'oneQuoted' => ['a="test"',null, null, ['a' => 'test']],
+            'oneQuotedEmpty' => ['a=""',null, null, ['a' => '']],
+            'oneSingleQuoted' => ["a='a'",null, null, ['a' => "'a'"]],
+            'mixedQuotes' => [
+                'a="tes\'t";test="\'test\'";test2="\\"quoted\\" test"',null, null,
+                ['a' => 'tes\'t', 'test' => '\'test\'', 'test2' => '"quoted" test']
+            ],
+            'slashes' => [
+                'a="\\t\\e\\s\\t";b="te\\\\st";c="\\"\\"',null, null,
+                ['a' => 'test', 'b' => 'te\\st', 'c' => '"\\']
+            ],
+            'specChars' => ['*=test;test=*',null, null, ['*' => 'test', 'test' => '*']],
+            'specChars2' => ['param*1=a;param*2=b',null, null, ['param*1' => 'a', 'param*2' => 'b']],
+            'numbers' => ['a=8888;b="999"',null, null, ['a' => '8888', 'b' => '999']],
+            'invalidQuotes2' => ['a="test',null, null, null, \InvalidArgumentException::class],
+            'invalidQuotes3' => ['a=test"',null, null, null, \InvalidArgumentException::class],
+            'invalidQuotes4' => ['a=te"st',null, null, null, \InvalidArgumentException::class],
+            'invalidEmptyValue' => ['a=b; c=',null, null, null, \InvalidArgumentException::class],
+            'invalidEmptyParam' => ['a=b; ;c=d',null, null, null, \InvalidArgumentException::class],
+            'semicolonAtEnd' => ['a=b;',null, null, null, \InvalidArgumentException::class],
+            'comma' => ['a=test,test',null, null, null, \InvalidArgumentException::class],
+
+            # true syntax
+            // 'spaces2' => [' a = b ; c = "d" ',null, null, ['a' => 'b', 'c' => 'd']],
```

Comment:
I can't find an RFC that specifically this case in general, but as I see it, spaces are not allowed on many points in [RFC2616](https://www.rfc-editor.org/rfc/rfc2616.html).

Reply:
It is better to check https://tools.ietf.org/html/rfc7230#section-3.2, 2616 is obsolete and less definitive.

```
 header-field   = field-name ":" OWS field-value OWS

     field-name     = token
     field-value    = *( field-content / obs-fold )
     field-content  = field-vchar [ 1*( SP / HTAB ) field-vchar ]
     field-vchar    = VCHAR / obs-text

     obs-fold       = CRLF 1*( SP / HTAB )
                    ; obsolete line folding
                    ; see Section 3.2.4

    obs-text       = %x80-FF

     OWS            = *( SP / HTAB )
               ; optional whitespace
```

Reply:
modified, optional spaces are allowed